app/cdap/components/DirectiveInput/Components/AutoComplete/index.tsx

```diff
@@ -0,0 +1,236 @@
+        });
+      }
+    }
+  };
+
+  const handleTabKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (input.length === 0 || input.split(' ').length !== 1) {
+      return;
+    }
+    event.preventDefault();
+    handleEnterKey();
+  };
+
+  useEffect(() => {
+    if (isDirectiveSelected) {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+    searchMatch(directiveInput);
+    setInput(directiveInput);
+  }, [directiveInput]);
+
+  const searchMatch = (query: string) => {
+    let results = [];
+    const input: string = query;
+    const spaceIndex: number = input.indexOf(' ');
+    if (fuse && input.length > 0) {
+      if (!isDirectiveSelected) {
+        results = fuse
+          .search(input)
+          .slice(0, 3)
+          .filter((row, index) => {
+            if (spaceIndex === -1) {
+              return true;
+            }
+            return row.score === 0 && index === 0;
+          })
+          .map((row) => {
+            row.uniqueId = uuidV4();
+            return row;
+          });
+        reverse(results);
+      } else {
+        results = fuse.search(input.split(':')[1]).map((row) => {
+          row.uniqueId = uuidV4();
+          return row;
+        });
+        reverse(results);
+      }
+    }
+    setActiveResults(results);
+    setInput(query);
+    setActiveSelectionIndex(results.length - 1);
+    if (!isDirectiveSelected) {
+      if ((spaceIndex !== -1) === true) {
+        getDirectiveUsage(results, true);
+      } else {
+        getDirectiveUsage(results, false);
+      }
+    }
+  };
+
+  const handleRowClick = (row) => {
+    if (typeof onRowClick !== 'function') {
+      return;
+    }
+    let eventObject = {} as IOnRowClickValue;
+    if (!isDirectiveSelected) {
+      eventObject = {
+        target: { value: `${row.item.directive}` },
+      };
+      onRowClick(eventObject);
+      getDirectiveUsage([row], true);
+    } else {
+      const splitData = input.split(/(?=[:])|(?<=[:])/g);
+      eventObject = {
+        target: { value: `${splitData[0]}${splitData[1]}${row.item.label}` },
+      };
+      setInput(`${splitData[0]}${splitData[1]}${row.item.label}`);
+      onRowClick(eventObject);
+      onColumnSelected(true);
+    }
+  };
+
+  return (
+    <Box>
+      {Array.isArray(activeResults) &&
+        activeResults.length > 0 &&
+        activeResults.map((row, index) => {
+          return (
+            <DirectiveResultRow
+              isActive={index === activeSelectionIndex}
+              key={row.uniqueId}
+              onClick={() => handleRowClick(row)}
+              data-testid="select-directive-list-option"
+            >
+              <Box>
+                <DirectiveLabel data-testid="select-directive-list-label" variant="body1">
+                  {row?.item?.directive || row?.item?.label}
+                </DirectiveLabel>
+                <DirectiveLabel
+                  isDescription={true}
```

Comment:
why ={true} again?
just write isDescription

Reply:
Removed not needed anymore